blockchain/state/state_object.go

```diff
@@ -197,14 +199,53 @@ func (self *stateObject) GetCommittedState(db Database, key common.Hash) common.
 		return value
 	}
 	// Track the amount of time wasted on reading the storage trie
+	var (
+		enc   []byte
+		err   error
+		meter *time.Duration
+	)
+	readStart := time.Now()
 	if EnabledExpensive {
-		defer func(start time.Time) { self.db.StorageReads += time.Since(start) }(time.Now())
+		// If the snap is 'under construction', the first lookup may fail. If that
+		// happens, we don't want to double-count the time elapsed. Thus this
+		// dance with the metering.
+		defer func(start time.Time) {
+			if meter != nil {
+				*meter += time.Since(readStart)
+			}
+		}(time.Now())
 	}
-	// Load from DB in case it is missing.
-	enc, err := self.getStorageTrie(db).TryGet(key[:])
-	if err != nil {
-		self.setError(err)
-		return common.Hash{}
+	if self.db.snap != nil {
+		if EnabledExpensive {
+			meter = &self.db.SnapshotStorageReads
+		}
+		// If the object was destructed in *this* block (and potentially resurrected),
+		// the storage has been cleared out, and we should *not* consult the previous
+		// snapshot about any storage values. The only possible alternatives are:
+		//   1) resurrect happened, and new slot values were set -- those should
+		//      have been handles via pendingStorage above.
+		//   2) we don't have new values, and can deliver empty response back
+		if _, destructed := self.db.snapDestructs[self.addrHash]; destructed {
+			return common.Hash{}
+		}
+		enc, err = self.db.snap.Storage(self.addrHash, crypto.Keccak256Hash(key.Bytes()))
+	}
+	// If the snapshot is unavailable or reading from it fails, load from the database.
+	if self.db.snap == nil || err != nil {
+		if EnabledExpensive {
+			if meter != nil {
+				// If we already spent time checking the snapshot, account for it
+				// and reset the readStart
+				*meter += time.Since(readStart)
+				readStart = time.Now()
+			}
+			meter = &self.db.StorageReads
+		}
```

Comment:
Ethereum code is like below.
```
		if meter != nil {
			// If we already spent time checking the snapshot, account for it
			// and reset the readStart
			*meter += time.Since(readStart)
			readStart = time.Now()
		}
		if metrics.EnabledExpensive {
			meter = &s.db.StorageReads
		}
```

Can I ask why you change like this?

Reply:
I have updated since I agreed with this comment https://github.com/klaytn/klaytn/pull/1155#discussion_r804370332.